src/Juvix/Backends/ArithmeticCircuit/Compilation.hs

```diff
@@ -1,31 +1,16 @@
-module Juvix.Backends.ArithmeticCircuit.Compilation where
+module Juvix.Backends.ArithmeticCircuit.Compilation (compile, runCirc, add, mul, sub, neg, eq, exp, int, c, and', or', Term, Type, lambda, var, input, cond) where
 
-import qualified Bulletproofs.ArithmeticCircuit as C
-import Data.Curve.Weierstrass.SECP256K1 (Fr)
+import Juvix.Library hiding (Type, exp)
 import Juvix.Backends.ArithmeticCircuit.Compilation.Types
-import Juvix.Backends.ArithmeticCircuit.Parameterisation
-import Juvix.Library hiding (Type)
+import qualified Circuit
+import qualified Circuit.Expr as Expr
+import qualified Juvix.Backends.ArithmeticCircuit.Parameterisation as Par
 
-compile ::
-  forall m.
-  (Monad m) =>
-  Term ->
-  Type ->
-  m (C.ArithCircuit Fr)
-compile _ _ = do
-  pure (flattenCircuit undefined)
+compile :: Term -> Type -> Circuit.ArithCircuit Par.F
+compile term _ =
+  let (_, circ) = transTerm mempty term
+  in case circ of
+    BoolExp exp -> Expr.execCircuitBuilder (Expr.compile exp)
+    FExp exp -> Expr.execCircuitBuilder (Expr.compile exp)
 
-flattenCircuit :: Circuit -> C.ArithCircuit Fr
-flattenCircuit _ =
-  C.ArithCircuit
-    { C.weights = C.GateWeights {C.wL = [], C.wR = [], C.wO = []},
-      C.commitmentWeights = [],
-      C.cs = []
-    }
-{-
- - find all output values
- - follow wire back to gate
- - if add/mul: add add/mul constraint of input values & output or intermediate value
- - if input: stop
- - otherwise: continue
- -}
+runCirc = Expr.execCircuitBuilder . Expr.compile
```

Comment:
Do we use this function? Looks like we could use it above.

Reply:
I use them in testcases to compile to `ArithCircuits`, should I move it to tests?

Reply:
If we only use it in test-cases, let's move it to the tests file, yeah